Code/Framework/AzCore/AzCore/DOM/DomPath.h

```diff
@@ -47,6 +47,10 @@ namespace AZ::Dom
         bool operator!=(const AZ::Name& key) const;
         bool operator!=(AZStd::string_view key) const;
 
+        //! Comparison operator used for storage in map structures
+        //! Compares hash order and not lexigraphic order, so not suitable for UI purposes
+        bool operator<(const PathEntry& rhs) const;
```

Comment:
It's good form to also add `operator>`, `operator<=` and `operator>=` when adding `operator<`.

Reply:
Ended up providing a hash instead for `unordered_map` so omitting for now